source/communication_interfaces/CMakeLists.txt

```diff
@@ -58,3 +71,35 @@ if(BUILD_TESTING)
     target_link_libraries(test_${PROJECT_NAME} ${PROJECT_NAME} ${GTEST_LIBRARIES} pthread)
     add_test(NAME test_${PROJECT_NAME} COMMAND test_${PROJECT_NAME})
 endif()
+
+
+# generate the version file for the config file
+write_basic_package_version_file(
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+        VERSION "${PROJECT_VERSION}"
+        COMPATIBILITY SameMajorVersion
+)
```

Comment:
nitpick: inconsistent spacing of closing parenthesis

Reply:
never mind, that's just how the formatting works when the arguments are listed on a new line. all good